### PR TITLE
[jit] Add a small optimization for unbox.any+stobj called with a null…

### PR DESCRIPTION
**Pull request by vargaz**
… on the stack.

### CODE REVIEW

File: mono/mini/method-to-ir.c [hunk 1]
```diff
@@ -11034,13 +11034,18 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
...
 				inline_costs += 2;
 			} else if (generic_class_is_reference_type (cfg, klass)) {
-				MONO_INST_NEW (cfg, res, OP_CASTCLASS);
-				res->dreg = alloc_preg (cfg);
-				res->sreg1 = (*sp)->dreg;
-				res->klass = klass;
-				res->type = STACK_OBJ;
-				MONO_ADD_INS (cfg->cbb, res);
-				cfg->flags |= MONO_CFG_HAS_TYPE_CHECK;
+				if ((*sp)->opcode == OP_PCONST && (*sp)->inst_p0 == 0) {
```
Comment (lewurm) [inline mono/mini/method-to-ir.c]: use the `MONO_INS_IS_PCONST_NULL` macro here

**Comment by lewurm** [comment]
is there a test that covers that scenario?

**Comment by vargaz** [comment]
Its hit a bunch of times by the normal test suite.

**Comment by kumpera** [comment]
Can we get a small test for it though? For mini

**Comment by kumpera** [comment]
LGTM

**Pull request merged**